File: server/bcephal-accounting-service/src/main/java/com/moriset/bcephal/accounting/domain/PostingBrowserData.java
/**
 * 
 */
package com.moriset.bcephal.accounting.domain;

import java.math.BigDecimal;
import java.util.Date;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.moriset.bcephal.domain.BrowserData;
import com.moriset.bcephal.utils.JsonDateDeserializer;
import com.moriset.bcephal.utils.JsonDateSerializer;

/**
 * 
 * @author MORISET-004
 *
 */
public class PostingBrowserData extends BrowserData{

	@JsonSerialize(using = JsonDateSerializer.class)
	@JsonDeserialize(using = JsonDateDeserializer.class)
	public Date valueDate;
	
	@JsonSerialize(using = JsonDateSerializer.class)
	@JsonDeserialize(using = JsonDateDeserializer.class)
	public Date entryDate;
	
	public String username;
	
	public String comment;
	
	public BigDecimal balance;
	
	@Enumerated(EnumType.STRING) 
	public PostingStatus status;
	
	
	public PostingBrowserData(Posting posting) {
		super(posting.getId(), posting.getName(), posting.getCreationDate(), posting.getModificationDate());	
		this.valueDate = posting.getValueDate();
		this.entryDate = posting.getEntryDate();
		this.username = posting.getUsername();
		this.balance = posting.computeBalance();
		this.status = posting.getStatus();
	}
	
}

File: client/bcephal-client-model/Accounting/PostingBrowserData.cs
using Bcephal.Models.Base;
using Newtonsoft.Json;
using System;

namespace Bcephal.Models.Accounting
{
    public class PostingBrowserData : BrowserData
    {

        public string valueDate;
        [JsonIgnore]
        public DateTime ValueDate
        {
            get
            {
                try
                {
                    return DateUtils.Parse(valueDate);
                }
                catch (Exception)
                {
                    return DateTime.Now;
                }
            }
            set
            {
                try
                {
                    valueDate = DateUtils.Format(value);
                }
                catch (Exception)
                {
                    valueDate = null;
                }
            }
        }

        public string entryDate;
        [JsonIgnore]
        public DateTime EntryDate
        {
            get
            {
                try
                {
                    return DateUtils.Parse(entryDate);
                }
                catch (Exception)
                {
                    return DateTime.Now;
                }
            }
            set
            {
                try
                {
                    entryDate = DateUtils.Format(value);
                }
                catch (Exception)
                {
                    entryDate = null;
                }
            }
        }

        public string comment { get; set; }

        public string username { get; set; }

        public string status { get; set; }
        [JsonIgnore]
        public PostingStatus Status
        {
            get { return !string.IsNullOrEmpty(status) ? PostingStatus.GetByCode(status) : null; }
            set { this.status = value != null ? value.code : null; }
        }

        public decimal? balance { get; set; }


    }
}
